engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -448,6 +450,14 @@ public void startApplication() throws YarnException, IOException
       }
       String libJarsCsv = copyFromLocal(fs, appPath, localJarFiles.toArray(new String[]{}));
 
+      if (UserGroupInformation.isSecurityEnabled() && customKeystoreConfig != null && customKeystorePath != null) {
+        String[] customSSLConfArr = {customKeystoreConfig, customKeystorePath};
+        String customSSLFileConf = copyFromLocal(fs, appPath, customSSLConfArr);
+        String configFileName = (new java.io.File(customKeystoreConfig)).getName();
+        dag.getAttributes().put(Context.DAGContext.STRAM_HTTP_CUSTOM_CONFIG, configFileName);
```

Comment:
Cryptic and convoluted naming is an issue and saying that due to "consistency" the same should be continued is what I consider wrong. Attributes and other elements introduced to the API need to be considered carefully as once they are in, they are very hard to change. And yes, not every committer is equally familiar with all areas of the project. If you think that naming in the API and user facing elements that are subject to backward compatibility isn't that important, then it is probably best to defer to those that think otherwise. I will request that this issue and other comments are addressed before this PR gets merged.